### ISSUE TITLE
Enhance L004 to enforce one indentation style across all files

### ISSUE THREAD

**Issue by NiallRees** [sqlfluff/sqlfluff#562]
Currently L004 throws an error if there is inconsistent indentation in a file (mixed spaces and tabs).

```
L: 121 | P:   1 | L004 | Mixed Tab and Space indentation found in file.
```

How would we feel about changing this to just enforce either tabs or spaces across all files? I personally would much prefer it as an anti-tabber, the logic would be simpler and a fix would be easy to implement. The indentation preference could just be set in the config.

**Comment by pwildenhain** [comment]
I think that makes sense. It doesn't sit right with me if one file has tabs, and another has spaces and I think `sqlfulff` should complain about it.
 
💡 What about adding an additional config option (e.g. `tabs_or_spaces`) under `sqlfluff:indentation`, and having it default to `spaces`? Then modifying `L004` to ensure that only that type of indentation exists in all files?

**Comment by NiallRees** [comment]
Do we still think we need `L002` with this fix? 
> Mixed Tabs and Spaces in single whitespace.

We now just enforce the indent style across all files so I think it's redundant. If we agree it's unneeded I will remove

**Comment by pwildenhain** [comment]
I agree we can remove, just need to make sure that we document that
breaking change right? What do you think is the best way to remove a rule?

**Comment by NiallRees** [comment]
I wonder if we should just leave it in but mark it as deprecated in favour of L004 in the docs?
